### PR TITLE
feat: add C implementation for `math/base/special/nanmax`

### PR DESCRIPTION
**Pull request by gunjjoshi**
## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for [`math/base/special/nanmax`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/nanmax).

```c
double stdlib_base_nanmax( const double x, const double y )
```

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #649.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/nanmax/README.md [hunk 1]
```diff
@@ -79,6 +79,130 @@ var v = nanmax( NaN, NaN );
...
+
+int main( void ) {
+    const double x[] = { 1.0, 0.45, -0.89, 0.0 / 0.0, -0.78, -0.22, 0.66, 0.11, -0.55, 0.0 };
+    const double y[] = { -0.22, 0.66, 0.0, -0.55, 0.33, 1.0, 0.0 / 0.0, 0.11, 0.45, -0.78 };
+
+    double v;
+    int i;
+    for ( i = 0; i < 10; i++ ) {
+        v = stdlib_base_nanmax( x[i], y[i] );
+        printf( "x[ %d ]: %f, y[ %d ]: %f, nanmax( x[ %d ], y[ %d ] ): %f\n", i, x[ i ], i, y[ i ], i, i, v );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/nanmax/README.md]:
```suggestion
        printf( "x[ %d ]: %lf, y[ %d ]: %lf, nanmax( x[ %d ], y[ %d ] ): %lf\n", i, x[ i ], i, y[ i ], i, i, v );
```

We're printing doubles, not floats.

Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/nanmax/README.md]: Thanks, updated this in `example.c` as well.

File: lib/node_modules/@stdlib/math/base/special/nanmax/test/test.native.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+});
+
+tape( 'the function returns the maximum value', opts, function test( t ) {
+	var v;
+
+	v = nanmax( 5.2, 3.14 );
+	t.strictEqual( v, 5.2, 'returns max value' );
+
+	v = nanmax( -4.2, 3.14 );
+	t.strictEqual( v, 3.14, 'returns max value' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/nanmax/test/test.native.js]:
```suggestion
	t.strictEqual( v, 5.2, 'returns expected value' );

	v = nanmax( -4.2, 3.14 );
	t.strictEqual( v, 3.14, 'returns expected value' );
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @gunjjoshi!

**Comment by kgryte** [comment]
As the lint failure does not stem from changes introduced in this PR, I'll go ahead and merge.

**Pull request merged**